test/jdk/java/net/httpclient/PlainConnectionLockTest.java

```diff
@@ -0,0 +1,382 @@
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions;
+
+import static java.net.http.HttpClient.Builder.NO_PROXY;
+import static java.net.http.HttpClient.Version.HTTP_1_1;
+
+/*
+ * @test id=default
+ * @bug 8372198
+ * @requires os.family != "windows"
+ * @summary Attempt to check that no deadlock occurs when
+ *          connections are closed by the ConnectionPool.
+ * @library /test/lib /test/jdk/java/net/httpclient/lib
+ * @build jdk.test.lib.net.SimpleSSLContext
+ *        jdk.httpclient.test.lib.common.HttpServerAdapters
+ * @run junit/othervm
+ *              -Djdk.httpclient.HttpClient.log=errors
+ *              -Djdk.httpclient.connectionPoolSize=1
+ *              ${test.main.class}
+ */
+/*
+ * @test id=windows
+ * @bug 8372198
+ * @requires os.family == "windows"
+ * @summary Attempt to check that no deadlock occurs when
+ *          connections are closed by the ConnectionPool.
+ * @library /test/lib /test/jdk/java/net/httpclient/lib
+ * @build jdk.test.lib.net.SimpleSSLContext
+ *        jdk.httpclient.test.lib.common.HttpServerAdapters
+ * @comment A special jtreg id for windows allows for experimentation
+ *          with different configuration - for instance, specifying
+ *          -Djdk.internal.httpclient.tcp.selector.useVirtualThreads=<always|never>
+ *          on the run command line, or specifying a different request count
+ *          with -DrequestCount=<integer>.
+ *          On windows, it seems important to set the backlog for the HTTP/1.1
+ *          server to at least the number of concurrent request. This is done
+ *          in the beforeTest() method.
+ *          If the test fails waiting for avalaible permits, due to system limitations,
+ *          even with the backlog correctly configure, adding a margin to the backlog
+ *          or reducing the requestCount could be envisaged.
+ * @run junit/othervm
+ *              -Djdk.httpclient.HttpClient.log=errors
+ *              -Djdk.httpclient.connectionPoolSize=1
+ *              ${test.main.class}
+ */
+
+// -Djava.security.debug=all
+class PlainConnectionLockTest implements HttpServerAdapters {
+
+    private SSLContext sslContext;
+    private HttpTestServer http1Server;
+    private HttpTestServer https1Server;
+    private String http1URI;
+    private String https1URI;
+    private ExecutorService serverExecutor;
+    private Semaphore responseSemaphore;
+    private Semaphore requestSemaphore;
+    private boolean successfulCompletion;
+    private static final int MANY = Integer.getInteger("requestCount", 100);
+
+    static {
+        HttpServerAdapters.enableServerLogging();
+    }
+
+    private boolean blockResponse(Semaphore request, Semaphore response) {
+        try {
+            request.release();
+            response.acquire();
+            return true;
+        } catch (InterruptedException x) {
```

Comment:
_Nit:_ While it might not make much of difference in this particular case, I think it is always a good practice to not discard `InterruptedException` – this has recently been [discussed on core-libs-dev].

```suggestion
        } catch (InterruptedException x) {
            Thread.currentThread().interrupt();  // Restore the interrupt
```

[discussed on core-libs-dev]: https://mail.openjdk.org/pipermail/core-libs-dev/2025-November/154551.html